src/validation.cpp

```diff
@@ -2203,15 +2204,18 @@ class ConnectTrace {
 
     void NotifyEntryRemoved(CTransactionRef txRemoved, MemPoolRemovalReason reason) {
         if (reason == MemPoolRemovalReason::CONFLICT) {
-            conflictedTxs.push_back(txRemoved);
+            conflictedTxs.back().push_back(txRemoved);
         }
     }
 
     void CallSyncTransactionOnConflictedTransactions() {
-        for (const auto& tx : conflictedTxs) {
-            GetMainSignals().SyncTransaction(*tx, NULL, CMainSignals::SYNC_TRANSACTION_NOT_IN_BLOCK);
+        for (const auto& txRemovedForBlock : conflictedTxs) {
+            for (const auto& tx : txRemovedForBlock) {
+                GetMainSignals().SyncTransaction(*tx, NULL, CMainSignals::SYNC_TRANSACTION_NOT_IN_BLOCK);
+            }
         }
         conflictedTxs.clear();
```

Comment:
Hmm, its removed by the end and there is no equivalent vector on which to do so, so I think its fine.